### ISSUE TITLE
no-single-line-block-comment needs to allow JSDoc

### ISSUE THREAD

**Issue by mohsen1** [microsoft/tslint-microsoft-contrib#385]
Code like 

```ts
/** Distance from earth to moon, in miles */
const DISTANCE = 238900;
```

...is very valid and there is no point of breaking that comment into multiple lines. 

Using a `//` comment will not carry over the comment in JSDoc.

**Comment by JoshuaKGoldberg** [comment]
@mohsen1 I actually like that feature of the rule. I assume the goal of the rule was to keep consistency with other block comments in where the comments start vertically:

```typescript
// Column 5
/** Distance from earth to moon, in miles */
const DISTANCE = 238900;

// Column 4
/**
 * Description.
 *
 * @returns Some value.
 */
const otherCode = () => boolean;
```

OTHO, the preference to save a couple of lines is valid as well. I think we should add a configuration to the rule for whether it should allow these single-line JSDoc comments. If it doesn't allow them, the error text should indicate to split across lines for their case.

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
